TokenMgrError Error not caught

There is a bug that causes the server to crash (on docker at least), here is the log from docker:

```java
17:58:36.429 [main] DEBUG io.netty.util.internal.logging.InternalLoggerFactory - Using SLF4J as the default logging framework
17:58:36.460 [main] DEBUG io.netty.util.ResourceLeakDetector - -Dio.netty.leakDetection.level: simple
17:58:36.467 [main] DEBUG io.netty.util.ResourceLeakDetector - -Dio.netty.leakDetection.maxRecords: 4
17:58:36.607 [main] INFO ratpack.server.RatpackServer - Starting server...
17:58:37.818 [main] DEBUG io.netty.util.internal.PlatformDependent0 - java.nio.Buffer.address: available
17:58:37.821 [main] DEBUG io.netty.util.internal.PlatformDependent0 - sun.misc.Unsafe.theUnsafe: available
17:58:37.823 [main] DEBUG io.netty.util.internal.PlatformDependent0 - sun.misc.Unsafe.copyMemory: available
17:58:37.824 [main] DEBUG io.netty.util.internal.PlatformDependent0 - java.nio.Bits.unaligned: true
17:58:37.826 [main] DEBUG io.netty.util.internal.PlatformDependent - Java version: 8
17:58:37.827 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.noUnsafe: false
17:58:37.828 [main] DEBUG io.netty.util.internal.PlatformDependent - sun.misc.Unsafe: available
17:58:37.829 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.noJavassist: false
17:58:38.073 [main] DEBUG io.netty.util.internal.PlatformDependent - Javassist: available
17:58:38.075 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.tmpdir: /tmp (java.io.tmpdir)
17:58:38.079 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.bitMode: 64 (sun.arch.data.model)
17:58:38.080 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.noPreferDirect: false
17:58:38.090 [main] DEBUG io.netty.util.internal.NativeLibraryLoader - -Dio.netty.tmpdir: /tmp (java.io.tmpdir)
17:58:38.091 [main] DEBUG io.netty.util.internal.NativeLibraryLoader - -Dio.netty.native.workdir: /tmp (io.netty.tmpdir)
17:58:38.142 [main] DEBUG io.netty.util.NetUtil - Loopback interface: lo (lo, 0:0:0:0:0:0:0:1%lo)
17:58:38.146 [main] DEBUG io.netty.util.NetUtil - /proc/sys/net/core/somaxconn: 128
17:58:38.182 [main] DEBUG io.netty.channel.MultithreadEventLoopGroup - -Dio.netty.eventLoopThreads: 2
17:58:38.301 [main] INFO ratpack.server.RatpackServer - Building registry...
17:58:38.518 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.numHeapArenas: 1
17:58:38.519 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.numDirectArenas: 1
17:58:38.521 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.pageSize: 8192
17:58:38.521 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.maxOrder: 11
17:58:38.522 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.chunkSize: 16777216
17:58:38.528 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.tinyCacheSize: 512
17:58:38.530 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.smallCacheSize: 256
17:58:38.531 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.normalCacheSize: 64
17:58:38.532 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.maxCachedBufferCapacity: 32768
17:58:38.533 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.cacheTrimInterval: 8192
17:58:39.268 [main] DEBUG io.netty.channel.DefaultChannelId - -Dio.netty.processId: 5 (auto-detected)
17:58:39.272 [main] DEBUG io.netty.channel.DefaultChannelId - -Dio.netty.machineId: 02:42:ac:ff:fe:11:00:03 (auto-detected)
17:58:39.273 [main] DEBUG io.netty.util.internal.ThreadLocalRandom - -Dio.netty.initialSeedUniquifier: 0x920f3a39e3d52c00 (took 86 ms)
17:58:39.322 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.allocator.type: pooled
17:58:39.323 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.threadLocalDirectBufferSize: 65536
17:58:39.323 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.maxThreadLocalCharBufferSize: 16384
17:58:39.345 [main] INFO ratpack.server.RatpackServer - Ratpack started (development) for http://localhost:4567
18:53:41.229 [ratpack-compute-1-2] DEBUG io.netty.buffer.AbstractByteBuf - -Dio.netty.buffer.bytebuf.checkAccessible: true
18:53:41.278 [ratpack-compute-1-2] DEBUG io.netty.util.internal.JavassistTypeParameterMatcherGenerator - Generated: io.netty.util.internal.__matchers__.io.netty.handler.codec.http.HttpRequestMatcher
18:53:41.290 [ratpack-compute-1-2] DEBUG io.netty.util.internal.JavassistTypeParameterMatcherGenerator - Generated: io.netty.util.internal.__matchers__.io.netty.handler.codec.http.HttpObjectMatcher
18:53:42.636 [ratpack-compute-1-2] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.maxCapacity: 262144
18:53:42.944 [ratpack-compute-1-2] DEBUG io.netty.handler.codec.compression.ZlibCodecFactory - -Dio.netty.noJdkZlibDecoder: false
18:53:42.946 [ratpack-compute-1-2] DEBUG io.netty.handler.codec.compression.ZlibCodecFactory - -Dio.netty.noJdkZlibEncoder: false
18:53:42.949 [ratpack-compute-1-2] DEBUG io.netty.util.internal.JavassistTypeParameterMatcherGenerator - Generated: io.netty.util.internal.__matchers__.io.netty.buffer.ByteBufMatcher
19:01:36.537 [ratpack-compute-1-1] ERROR ratpack.error.internal.DefaultDevelopmentErrorHandler - exception thrown for request to /search
java.util.concurrent.CompletionException: com.github.javaparser.TokenMgrError: Lexical error at line 1, column 1.  Encountered: "#" (35), after : ""
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:604)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1595)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.github.javaparser.TokenMgrError: Lexical error at line 1, column 1.  Encountered: "#" (35), after : ""
	at com.github.javaparser.ASTParserTokenManager.getNextToken(ASTParserTokenManager.java:2480)
	at com.github.javaparser.ASTParser.jj_scan_token(ASTParser.java:9115)
	at com.github.javaparser.ASTParser.jj_3R_217(ASTParser.java:8226)
	at com.github.javaparser.ASTParser.jj_3R_181(ASTParser.java:8284)
	at com.github.javaparser.ASTParser.jj_3R_127(ASTParser.java:8328)
	at com.github.javaparser.ASTParser.jj_3R_126(ASTParser.java:7055)
	at com.github.javaparser.ASTParser.jj_3R_76(ASTParser.java:7068)
	at com.github.javaparser.ASTParser.jj_3_1(ASTParser.java:7144)
	at com.github.javaparser.ASTParser.jj_2_1(ASTParser.java:5170)
	at com.github.javaparser.ASTParser.CompilationUnit(ASTParser.java:146)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:111)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:133)
	at xyz.exemplator.exemplator.parser.java.JavaParser.of(JavaParser.java:38)
	at xyz.exemplator.exemplator.parser.Parsers.from(Parsers.java:23)
	at xyz.exemplator.exemplator.Router.parseOrReturn(Router.java:116)
	at xyz.exemplator.exemplator.Router.lambda$null$3(Router.java:81)
	at java.util.Optional.flatMap(Optional.java:241)
	at xyz.exemplator.exemplator.Router.lambda$null$4(Router.java:80)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
	... 6 common frames omitted
19:02:46.922 [ratpack-compute-1-2] ERROR ratpack.error.internal.DefaultDevelopmentErrorHandler - 405 client error for request to /
19:02:48.328 [ratpack-compute-1-2] ERROR ratpack.error.internal.DefaultDevelopmentErrorHandler - 404 client error for request to /favicon.ico
19:04:00.848 [ratpack-compute-1-1] ERROR ratpack.error.internal.DefaultDevelopmentErrorHandler - 405 client error for request to /
19:04:02.040 [ratpack-compute-1-1] ERROR ratpack.error.internal.DefaultDevelopmentErrorHandler - 404 client error for request to /favicon.ico
19:04:12.872 [ratpack-compute-1-2] ERROR ratpack.error.internal.DefaultDevelopmentErrorHandler - exception thrown for request to /search
java.util.concurrent.CompletionException: com.github.javaparser.TokenMgrError: Lexical error at line 1, column 1.  Encountered: "#" (35), after : ""
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:604)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1595)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.github.javaparser.TokenMgrError: Lexical error at line 1, column 1.  Encountered: "#" (35), after : ""
	at com.github.javaparser.ASTParserTokenManager.getNextToken(ASTParserTokenManager.java:2480)
	at com.github.javaparser.ASTParser.jj_scan_token(ASTParser.java:9115)
	at com.github.javaparser.ASTParser.jj_3R_217(ASTParser.java:8226)
	at com.github.javaparser.ASTParser.jj_3R_181(ASTParser.java:8284)
	at com.github.javaparser.ASTParser.jj_3R_127(ASTParser.java:8328)
	at com.github.javaparser.ASTParser.jj_3R_126(ASTParser.java:7055)
	at com.github.javaparser.ASTParser.jj_3R_76(ASTParser.java:7068)
	at com.github.javaparser.ASTParser.jj_3_1(ASTParser.java:7144)
	at com.github.javaparser.ASTParser.jj_2_1(ASTParser.java:5170)
	at com.github.javaparser.ASTParser.CompilationUnit(ASTParser.java:146)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:111)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:133)
	at xyz.exemplator.exemplator.parser.java.JavaParser.of(JavaParser.java:38)
	at xyz.exemplator.exemplator.parser.Parsers.from(Parsers.java:23)
	at xyz.exemplator.exemplator.Router.parseOrReturn(Router.java:116)
	at xyz.exemplator.exemplator.Router.lambda$null$3(Router.java:81)
	at java.util.Optional.flatMap(Optional.java:241)
	at xyz.exemplator.exemplator.Router.lambda$null$4(Router.java:80)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
	... 6 common frames omitted
19:04:47.732 [ratpack-compute-1-2] ERROR ratpack.error.internal.DefaultDevelopmentErrorHandler - exception thrown for request to /search
java.util.concurrent.CompletionException: com.github.javaparser.TokenMgrError: Lexical error at line 1, column 1.  Encountered: "#" (35), after : ""
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:604)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1595)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.github.javaparser.TokenMgrError: Lexical error at line 1, column 1.  Encountered: "#" (35), after : ""
	at com.github.javaparser.ASTParserTokenManager.getNextToken(ASTParserTokenManager.java:2480)
	at com.github.javaparser.ASTParser.jj_scan_token(ASTParser.java:9115)
	at com.github.javaparser.ASTParser.jj_3R_217(ASTParser.java:8226)
	at com.github.javaparser.ASTParser.jj_3R_181(ASTParser.java:8284)
	at com.github.javaparser.ASTParser.jj_3R_127(ASTParser.java:8328)
	at com.github.javaparser.ASTParser.jj_3R_126(ASTParser.java:7055)
	at com.github.javaparser.ASTParser.jj_3R_76(ASTParser.java:7068)
	at com.github.javaparser.ASTParser.jj_3_1(ASTParser.java:7144)
	at com.github.javaparser.ASTParser.jj_2_1(ASTParser.java:5170)
	at com.github.javaparser.ASTParser.CompilationUnit(ASTParser.java:146)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:111)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:133)
	at xyz.exemplator.exemplator.parser.java.JavaParser.of(JavaParser.java:38)
	at xyz.exemplator.exemplator.parser.Parsers.from(Parsers.java:23)
	at xyz.exemplator.exemplator.Router.parseOrReturn(Router.java:116)
	at xyz.exemplator.exemplator.Router.lambda$null$3(Router.java:81)
	at java.util.Optional.flatMap(Optional.java:241)
	at xyz.exemplator.exemplator.Router.lambda$null$4(Router.java:80)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
```

Seems like the error comes from parsing illegal characters like ```#``` here.

--- a/src/main/java/xyz/exemplator/exemplator/parser/java/JavaParser.java
+++ b/src/main/java/xyz/exemplator/exemplator/parser/java/JavaParser.java
@@ -1,6 +1,7 @@
 package xyz.exemplator.exemplator.parser.java;
 
 import com.github.javaparser.ParseException;
+import com.github.javaparser.TokenMgrError;
 import com.github.javaparser.ast.CompilationUnit;
 import com.github.javaparser.ast.ImportDeclaration;
 import com.github.javaparser.ast.body.TypeDeclaration;
@@ -36,7 +37,7 @@ public class JavaParser implements Parser {
         try {
             // parse the file
             cu = com.github.javaparser.JavaParser.parse(inputStream);
-        } catch (ParseException e) {
+        } catch (ParseException | TokenMgrError e) {
             logger.error("Unable to parse input", e);
             return Optional.empty();
         } finally {

